pandas_gbq/gbq.py

```diff
@@ -847,44 +738,31 @@ def _parse_data(schema, rows):
     fields = schema['fields']
     col_types = [field['type'] for field in fields]
     col_names = [str(field['name']) for field in fields]
-    col_dtypes = [dtype_map.get(field['type'], object) for field in fields]
+    col_dtypes = [
+        dtype_map.get(field['type'].upper(), object)
+        for field in fields
+    ]
+    print(fields)
     page_array = np.zeros((len(rows),), dtype=lzip(col_names, col_dtypes))
-    for row_num, raw_row in enumerate(rows):
-        entries = raw_row.get('f', [])
-        for col_num, field_type in enumerate(col_types):
-            field_value = _parse_entry(entries[col_num].get('v', ''),
-                                       field_type)
+    for row_num, entries in enumerate(rows):
+        for col_num in range(len(col_types)):
+            field_value = entries[col_num]
```

Comment:
Because this is being called so many times, you may even get a small speed up from eliminating assigning to `field_value`

(but these are all things that are either the same or better than the existing version)